docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,326 @@
+object is created in the `garden` namespace, and drives the seed's `prometheus-operator` to configure a scrape on that
+endpoint by `prometheus-seed`.
+
+`pvc-autoscaler` runs in active/passive replication mode, based on the standard leader election mechanism, supplied by
+the K8s controller runtime library. Only the active replica is shown in _Fig.1_.
+
+### PVC Autoscaler - Deployer Structure
+The `pvc-autoscaler` application and its supporting artefacts are deployed by `gardener`, as part of the seed
+reconciliation flow.
+
+The `vali` and `prometheus` deployers, part of `gardenlet`'s shoot reconciliation flow, ensure that the PVC template section in
+the respective StatefulSet objects contains the annotations, necessary to instruct `pvc-autuscaler` how to act on the
+PVCs, created from those templates.
+
+That `gardenlet` action mechanism, based on PVC templates, only affects PVCs upon creation. Therefore, it has no effect
+on existing shoots, and on the observability PVCs of existing seed clusters. To enable PVC autoscaling for existing
+shoots, those PVCs will be directly annotated by the operator. Such direct intervention allows gradual transition,
+where autoscaling can initially be enabled only for a few PVCs for validation purposes.
+Similarly, to enable PVC autoscaling on existing seed clusters' `garden` PVCs, those will be directly annotated at
+operators' discretion.
+
+![02-operator_structure.png](assets/gep29-02-operator_structure.png)
+
+_Fig.2: Operator structure_
+
+VPA is used to scale `pvc-autoscaler`. For the future security enhancement scenario, where `kube-rbac-proxy` is used
+(see [section 'Future Enhancements'](#future-enhancements)),
+that container is expected to be excluded from autoscaling, and to run with fixed resource requests. Pod evictions
+driven by that lean container are unjustified, as they would only achieve minimal resource savings, and disrupt
+the main container.
+
+The feature is deployed behind a dedicated feature gate. When disabled, `pvc-autoscaler` is removed, and related
+annotations are removed from the shoot control plane observability StatefulSets, and the seed observability StatefulSets.
+
+#### Initial volume size and MaxAllowed
+
+| Component type | Component            | Old size | Initial size    | Max size |
+|----------------|----------------------|----------|-----------------|----------|
+| Shoot          | prometheus           | 20Gi     | 1Gi             | 40Gi     |
+| Shoot          | vali                 | 30Gi     | 1Gi             | 60Gi     |
+| Seed           | prometheus-aggregate | 20Gi     | 2Gi             | 40Gi     |
+| Seed           | prometheus-cache     | 10Gi     | 6Gi<sup>*</sup> | 20Gi     |
+| Seed           | prometheus-seed      | 100Gi    | 5Gi             | 200Gi    |
+| Seed           | vali                 | 100Gi    | 5Gi             | 200Gi    |
```

Comment:
Where are the max sizes coming from?

Reply:
It's just 2x current size. I used a very sound scientific approach: I more or less closed my eyes and tossed a coin. I'm absolutely open to better suggestions. The factors that affected my choice are:
- I want to allow the autoscaler to provision more than current storage, where needed.
- With an autoscaler that's not combat-proven, I'm concerned that it might have a bug which causes it to unnecessarily expand volumes all the way to the max size. And there's no support for shrinking. So I'm worried about the scenario where we're forced to delete all observability volumes, just so we can restore them to acceptable size.

So I allowed the autoscaler the freedom up to 2x current size. My thinking is: 2x should suffice for a couple of months. And after that time, we'll have gained confidence in the reliability of `pvc-autoscaler` and will be comfortable increasing the max value to its ultimate level.